app/src/components/common/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { trackErrorBoundaryShown } from "modules/analytics/events/common/error-boundaries";
+
+type Props = {
+  fallback: React.ReactNode;
+  children?: React.ReactNode;
+};
+
+type State = {
+  isError: boolean;
+};
+
+export class ErrorBoundary extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { isError: false };
+  }
+
+  static getDerivedStateFromError(error: unknown) {
+    return { isError: true };
+  }
+
+  componentDidCatch(error: unknown, errorInfo: unknown) {
+    trackErrorBoundaryShown(error, errorInfo);
```

Comment:
I think this is too much information to log. Stack trace can be too many lines.
`error.message` contains the high-level message which we can log. And maybe first 100 chars from stack trace will contain the context.